### PR TITLE
[OB3] Improvements and Changes to Consent Manager Self-Care Portal UI

### PR DESCRIPTION
**Pull request by Sandaru-Vithanage**
## [OB3] Improvements and Changes to Consent Manager Self-Care Portal UI

> This pull request contains changes to UI of consent manager self-care portal, relevant changes have been done in order to rectify the issues of https://github.com/wso2/financial-open-banking/issues/48 
Following changes to the UI of Consent Manager Self Care Portal has been done.
>1. Introduces Grid layout for the key dates info tab 
>2. Fixed screen responsive accounts info tab
>3. Fixed Download confirmation of consent link
>4. Fixed shadows of the button

>Key Dates and Account Info (Before)
>newicon

>Key Dates and Account Info (After Changes)
>Screenshot 2024-03-07 142645

>Profile Details Container (Before)
>Screenshot 2024-01-31 134636

>Profile Details Container (After)
Screenshot 2024-03-07 143059

**Issue link:** https://github.com/wso2/financial-open-banking/issues/48

**Doc Issue:** *Optional, link issue from [documentation repository](https://github.com/wso2/docs-ob/issues)*

**Applicable Labels:** OB 3.0.0

------

### Development Checklist

1. [ ] Build complete solution with pull request in place.
2. [ ] Ran checkstyle plugin with pull request in place.
3. [ ] Ran Findbugs plugin with pull request in place.
4. [ ] Ran FindSecurityBugs plugin and verified report.
5. [ ] Formatted code according to WSO2 code style.
6. [ ] Have you verified the PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?
7. [ ] Migration scripts written (if applicable).
8. [ ] Have you followed secure coding standards in [WSO2 Secure Engineering Guidelines](http://wso2.com/technical-reports/wso2-secure-engineering-guidelines)?

### Testing Checklist

1. [ ] Written unit tests.
2. [ ] Verified tests in multiple database environments (if applicable).
3. [ ] Tested with BI enabled  (if applicable).

### CODE REVIEW

File: react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/specConfigs/common.js [hunk 1]
```diff
@@ -17,61 +17,58 @@
  */
 
 export const common = {
-    footerContent: "WSO2 Open Banking | 2021",
-    complaintHandleLinkText: "Complaint handling and resolution",
+  footerContent: 'WSO2 Open Banking | 2024',
```
Comment (Ashi1993) [inline react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/specConfigs/common.js:20]: Shall we retrieve the year from current date and add here

Comment (Sandaru-Vithanage) [reply react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/specConfigs/common.js]: Fixed this in [a80b0de](https://github.com/wso2/financial-open-banking/pull/49/commits/a80b0dee72da85a13bb6e2ca48216d1626320953)

File: react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/specConfigs/common.js [hunk 2]
```diff
@@ -17,7 +17,7 @@
  */
 
 export const common = {
-    footerContent: "WSO2 Open Banking | 2021",
+    footerContent: "WSO2 Open Banking | "  + new Date().getFullYear(),
```
Comment (VimukthiRajapaksha) [inline react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/specConfigs/common.js:20]:
```suggestion
    footerContent: "WSO2 Open Banking | " + new Date().getFullYear(),
```

File: react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/specConfigs/specConfigurations.js [hunk 3]
```diff
@@ -173,7 +173,7 @@ export const account_lang = [
             },
         ],
         profile: {
-            confirmation: "View confirmation of consent >"
+            confirmation: "'Download confirmation of consent'"
```
Comment (VimukthiRajapaksha) [inline react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/specConfigs/specConfigurations.js:176]:
```suggestion
            confirmation: "Download confirmation of consent"
```

File: react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/detailedAgreementPage/KeyDatesInfo.jsx [hunk 4]
```diff
@@ -127,7 +127,7 @@ export const KeyDatesInfo = ({consent, infoLabels, consentType}) => {
     return (
         <div className="keyDatesBody">
             <h5>{keyDatesConfig.keyDatesInfoLabel}</h5>
-            {keyDatesMap}
+            <div className="two-columns">{keyDatesMap}</div>
```
Comment (VimukthiRajapaksha) [inline react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/detailedAgreementPage/KeyDatesInfo.jsx:130]: Without manually defining styles, can't we use existing bootstrap breakpoints[1] for this?

[1] https://getbootstrap.com/docs/5.3/layout/breakpoints/#available-breakpoints

**Review by VimukthiRajapaksha — APPROVED** [review]

File: react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/api/consent-api.js [hunk 5]
```diff
@@ -170,3 +172,45 @@ export const getConsentsFromAPIForSearch = (searchObj, user, appInfo) => {
...
+    setWithdrawIconId,
+    setShow
+) => {
+    try {
+        const getRevokeUrl = (consentId, user) => {
+            const adminUrl = `${CONFIG.BACKEND_URL}/admin/revoke?consentID=${consentId}`;
+            const defaultUrl = `${CONFIG.BACKEND_URL}/admin/revoke?consentID=${consentId}&userID=${user.email}`;
+
+            return user.role === 'customerCareOfficer' ? adminUrl : defaultUrl;
+        };
```
Comment (VimukthiRajapaksha) [inline react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/api/consent-api.js:191]: Can we move the `getRevokeUrl` arrow function out of the `withdrawConsent` function?

File: react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/api/consent-api.js [hunk 6]
```diff
@@ -170,3 +172,45 @@ export const getConsentsFromAPIForSearch = (searchObj, user, appInfo) => {
         });
 };
 
+export const withdrawConsent = async (
```
Comment (VimukthiRajapaksha) [inline react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/api/consent-api.js:175]: Shall we rename this to revokeConsent?

File: react-apps/self-care-portal/self-care-portal-frontend/package.json [hunk 7]
```diff
@@ -24,6 +24,7 @@
     "react": "^17.0.1",
     "react-axios": "^2.0.6",
     "react-bootstrap": "^1.4.0",
+    "react-bootstrap-icons": "^1.11.3",
```
Comment (VimukthiRajapaksha) [inline react-apps/self-care-portal/self-care-portal-frontend/package.json:27]: Where do we use these icons?

File: react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/api/consent-api.js [hunk 8]
```diff
@@ -170,3 +172,45 @@ export const getConsentsFromAPIForSearch = (searchObj, user, appInfo) => {
...
+        });
+        if (response.status === 204) {
+            setMessage(withdrawLang.withdrawModalSuccessMsg + applicationName);
+            setWithdrawMessageIcon(faCheckCircle);
+            setWithdrawIconId('withdrawSuccess');
+        } else {
+            setMessage(withdrawLang.withdrawModalFailMsg);
+            setWithdrawMessageIcon(faExclamationCircle);
+            setWithdrawIconId('withdrawFail');
+        }
```
Comment (VimukthiRajapaksha) [inline react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/api/consent-api.js:209]: Shouldn't this function return a promise, similar to getConsentsFromAPI?

File: react-apps/self-care-portal/self-care-portal-frontend/package.json (hunk 7 above)
Comment (Sandaru-Vithanage) [reply react-apps/self-care-portal/self-care-portal-frontend/package.json:27]: This is used to import the `Download` icon in `profileMain.jsx`,

Comment (VimukthiRajapaksha) [reply react-apps/self-care-portal/self-care-portal-frontend/package.json:27]: Is it possible to use the download icon from the existing `fontawesome` library[1]?

[1] https://fontawesome.com/search?q=download&o=r

File: react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/api/consent-api.js (hunk 8 above)
Comment (Sandaru-Vithanage) [reply react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/api/consent-api.js]: changed the `revokeConsent` function to return a Promise and moved response handling logics to `WithdrawStep2.jsc` in [106de5c](https://github.com/wso2/financial-open-banking/pull/49/commits/106de5ca75429194a33fa5474cba8b5fbdc115ff)

File: react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/api/consent-api.js (hunk 5 above)
Comment (Sandaru-Vithanage) [reply react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/api/consent-api.js]: resolved in [106de5c](https://github.com/wso2/financial-open-banking/pull/49/commits/106de5ca75429194a33fa5474cba8b5fbdc115ff)

File: react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/api/consent-api.js (hunk 6 above)
Comment (Sandaru-Vithanage) [reply react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/api/consent-api.js]: resolved in [106de5c](https://github.com/wso2/financial-open-banking/pull/49/commits/106de5ca75429194a33fa5474cba8b5fbdc115ff)

**Review by Ashi1993 — APPROVED** [review]

File: react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/detailedAgreementPage/WithdrawStep2.jsx [hunk 9]
```diff
@@ -198,9 +159,9 @@ export const WithdrawStep2 = ({ match }) => {
             </div>
             <div className="actionBtnDiv">
               <button
-                onClick={withdrawConsent}
+              onClick={handleRevokeConsent}
```
Comment (hasithakn) [inline react-apps/self-care-portal/self-care-portal-frontend/accelerator/src/detailedAgreementPage/WithdrawStep2.jsx:162]: Check indentation

**Review by hasithakn — APPROVED** [review]

**Pull request merged**